python/sglang/srt/configs/model_config.py

```diff
@@ -46,10 +47,23 @@ def __init__(
             model_override_args=model_override_args,
         )
         self.hf_text_config = get_hf_text_config(self.hf_config)
+        derived_context_len = get_context_length(self.hf_text_config)
+        allow_long_context = os.environ.get("SGLANG_ALLOW_LONG_MAX_MODEL_LEN", None)
+        
         if context_length is not None:
-            self.context_len = context_length
+            if context_length > derived_context_len:
+                if allow_long_context:
+                    print(f"Warning: User-specified max_model_len ({context_length}) is greater than the derived max_model_len ({derived_context_len}). "
+                          f"This may lead to incorrect model outputs or CUDA errors.")
+                    self.context_len = context_length
+                else:
+                    raise ValueError(f"User-specified max_model_len ({context_length}) is greater than the derived max_model_len "
+                                     f"(max_position_embeddings={derived_context_len} or model_max_length=None in model's config.json). "
```

Comment:
I change it to:

```python
raise ValueError(f"User-specified context_length ({context_length}) is greater than the derived context_length ({derived_context_len}). "
                                     f"This may lead to incorrect model outputs or CUDA errors. Note that the derived context_length may differ from max_position_embeddings in the model's config. "
                                     f"To allow overriding this maximum, set the env var SGLANG_ALLOW_OVERWRITE_LONGER_CONTEXT_LEN=1")
```

Is this acceptable?

Reply:
sounds good.